src/goDebugConfiguration.ts

```diff
@@ -1,136 +1,147 @@
+		if (!debugConfiguration['env']) {
+			debugConfiguration['env'] = { GOPATH: gopath };
+		} else if (!debugConfiguration['env']['GOPATH']) {
+			debugConfiguration['env']['GOPATH'] = gopath;
+		}
+
+		const goConfig = getGoConfig(folder && folder.uri);
+		const goToolsEnvVars = getToolsEnvVars();
+		Object.keys(goToolsEnvVars).forEach((key) => {
+			if (!debugConfiguration['env'].hasOwnProperty(key)) {
+				debugConfiguration['env'][key] = goToolsEnvVars[key];
+			}
+		});
+
+		const dlvConfig: { [key: string]: any } = goConfig.get('delveConfig');
+		let useApiV1 = false;
+		if (debugConfiguration.hasOwnProperty('useApiV1')) {
+			useApiV1 = debugConfiguration['useApiV1'] === true;
+		} else if (dlvConfig.hasOwnProperty('useApiV1')) {
+			useApiV1 = dlvConfig['useApiV1'] === true;
+		}
+		if (useApiV1) {
+			debugConfiguration['apiVersion'] = 1;
+		}
+		if (!debugConfiguration.hasOwnProperty('apiVersion') && dlvConfig.hasOwnProperty('apiVersion')) {
+			debugConfiguration['apiVersion'] = dlvConfig['apiVersion'];
+		}
+		if (!debugConfiguration.hasOwnProperty('dlvLoadConfig') && dlvConfig.hasOwnProperty('dlvLoadConfig')) {
+			debugConfiguration['dlvLoadConfig'] = dlvConfig['dlvLoadConfig'];
+		}
+		if (
+			!debugConfiguration.hasOwnProperty('showGlobalVariables') &&
+			dlvConfig.hasOwnProperty('showGlobalVariables')
+		) {
+			debugConfiguration['showGlobalVariables'] = dlvConfig['showGlobalVariables'];
+		}
+		if (debugConfiguration.request === 'attach' && !debugConfiguration['cwd']) {
+			debugConfiguration['cwd'] = '${workspaceFolder}';
+		}
+
+		debugConfiguration['dlvToolPath'] = getBinPath('dlv');
+		if (!path.isAbsolute(debugConfiguration['dlvToolPath'])) {
+			promptForMissingTool('dlv');
+			return;
+		}
+
+		if (debugConfiguration['mode'] === 'auto') {
+			debugConfiguration['mode'] =
+				activeEditor && activeEditor.document.fileName.endsWith('_test.go') ? 'test' : 'debug';
+		}
+		debugConfiguration['currentFile'] =
+			activeEditor && activeEditor.document.languageId === 'go' && activeEditor.document.fileName;
+
+		this.showWarning(
+			'ignoreDebugLaunchRemoteWarning',
+			'Request type of \'launch\' with mode \'remote\' is deprecated, please use request type \'attach\' with mode \'remote\' instead.',
+			debugConfiguration.request === 'launch' && debugConfiguration['mode'] === 'remote');
+
+		this.showWarning(
+			'ignoreUsingRemotePathAndProgramWarning',
+			'Request type of \'attach\' with mode \'remote\' does not work with \'program\' attribute. Please use \'cwd\' attribute instead.',
+			debugConfiguration.request === 'attach' && debugConfiguration['mode'] === 'remote' && debugConfiguration['program']);
+
+		return debugConfiguration;
+	}
+
+	/**
+	 * Show warningMessage to the user if warningCondition is met.
+	 * If the user chooses not to see the warning again, sets it to the global state.
+	 */
+	private showWarning(warningKey: string, warningMessage: string, warningCondition: boolean) {
```

Comment:
If `warningCondition` is true, then the entire code path in this method is a no-op. It would be easier to read the code if this condition was checked outside and this function called only if it were true

Reply:
Also would suggest renaming `warningKey` to `ignoreWarningKey`